### PR TITLE
Feature: polymorphic allocator

### PR DESCRIPTION
**Pull request by Clueliss**

### CODE REVIEW

**Comment by Clueliss** [comment]
Not sure whats wrong with the conan/cmake file. But it doesn't find boost. Could you have a look at that @bigerl ?

File: tests/tests_polymorphic_allocator.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#include <doctest/doctest.h>
+
+#include <dice/template-library/polymorphic_allocator.hpp>
+
+#include <memory>
+#include <memory_resource>
+
+TEST_SUITE("polymorphic_allocator") {
+	template<typename T>
+	using poly_alloc_t = dice::template_library::polymorphic_allocator<int, std::allocator, std::pmr::polymorphic_allocator>;
```
Comment (bigerl) [inline tests/tests_polymorphic_allocator.cpp:11]: Why is the template parameter unused?

File: tests/tests_polymorphic_allocator.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		auto ptr = std::allocator_traits<alloc_t>::allocate(alloc, 1);
+		*ptr = 5;
+		CHECK(*ptr == 5);
+		std::allocator_traits<alloc_t>::deallocate(alloc, ptr, 1);
+	}
+#endif//DICE_TEMPLATE_LIBRARY_WITH_BOOST
+
+	TEST_CASE("polymorphic alloc") {
+		SUBCASE("with default alloc") {
+			poly_alloc_t<int> alloc{};
```
Comment (bigerl) [inline tests/tests_polymorphic_allocator.cpp:27]: T is int here anyway

**Review by bigerl — CHANGES_REQUESTED** [review]
Code looks good. Only a tiny thing in the tests. 

For merging into dice-template-library we also need:

- [ ] An example how to use it.
- [x] A section in the readme. 

I would suggest to base the readme around persistent memory. 
Maybe mention metall but actually use boost interprocess (because we have boost headers anyway).

File: tests/tests_polymorphic_allocator.cpp (hunk 1 above)
Comment (Clueliss) [reply tests/tests_polymorphic_allocator.cpp:11]: Good point

**Review by bigerl — APPROVED** [review]

**Pull request merged**